kafka/common_test.go

```diff
@@ -318,3 +328,160 @@ func TestTopicFieldFunc(t *testing.T) {
 		assert.Equal(t, zap.String("topic", "c"), topic)
 	})
 }
+
+// generateValidCACert creates a valid self-signed CA certificate in PEM format.
+func generateValidCACert(t testing.TB) ([]byte, *x509.Certificate, *rsa.PrivateKey) {
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
+	require.NoError(t, err)
+
+	template := x509.Certificate{
+		SerialNumber:          big.NewInt(int64(time.Now().Year())),
+		Subject:               pkix.Name{CommonName: "Test CA"},
+		NotBefore:             time.Now().Add(-time.Hour),
+		NotAfter:              time.Now().Add(time.Hour),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &key.PublicKey, key)
+	require.NoError(t, err)
+
+	return pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: derBytes}), &template, key
+}
+
+func TestTLSCACertPath(t *testing.T) {
+	t.Run("valid cert", func(t *testing.T) {
+		t.Setenv("KAFKA_PLAINTEXT", "") // clear plaintext mode
+
+		tempFile := filepath.Join(t.TempDir(), "ca_cert.pem")
+		validCert, _, _ := generateValidCACert(t)
+		err := os.WriteFile(tempFile, validCert, 0644)
+		require.NoError(t, err)
+
+		t.Setenv("KAFKA_TLS_CA_CERT_PATH", tempFile)
+		cfg := CommonConfig{Brokers: []string{"broker"}, Logger: zap.NewNop()}
+		require.NoError(t, cfg.finalize())
+		require.NotNil(t, cfg.Dialer)
+		require.Nil(t, cfg.TLS)
+	})
+	t.Run("missing file", func(t *testing.T) {
+		t.Setenv("KAFKA_PLAINTEXT", "")
+		tempFile := filepath.Join(t.TempDir(), "nonexistent_cert.pem")
+		t.Setenv("KAFKA_TLS_CA_CERT_PATH", tempFile)
+		cfg := CommonConfig{Brokers: []string{"broker"}, Logger: zap.NewNop()}
+		err := cfg.finalize()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "kafka: error creating dialer with CA cert")
+		require.Contains(t, err.Error(), "no such file or directory")
+	})
+	t.Run("invalid cert", func(t *testing.T) {
+		t.Setenv("KAFKA_PLAINTEXT", "")
+		tempFile := filepath.Join(t.TempDir(), "invalid_cert.pem")
+		err := os.WriteFile(tempFile, []byte("invalid pem data"), 0644)
+		require.NoError(t, err)
+
+		t.Setenv("KAFKA_TLS_CA_CERT_PATH", tempFile)
+		cfg := CommonConfig{Brokers: []string{"broker"}, Logger: zap.NewNop()}
+		err = cfg.finalize()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "kafka: error creating dialer with CA cert")
+	})
+}
+
+func TestTLSHotReload(t *testing.T) {
```

Comment:
Just a thought: maybe this test could be simplified by using httptest? httptest.NewTLSServer creates its own certificate, and you can extract the CA out of the `Server.Client()`'s TLS config.

Reply:
Refactored and re-used the httptest.server TLS certificates now.